x-pack/platform/plugins/private/intercepts/server/core.ts

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import crypto from 'crypto';
+import {
+  RequestHandlerContext,
+  type CoreSetup,
+  type CoreStart,
+  type Logger,
+  type IRouter,
+  type IContextProvider,
+} from '@kbn/core/server';
+import { schema } from '@kbn/config-schema';
+import { parseIntervalAsMillisecond } from '@kbn/task-manager-plugin/server/lib/intervals';
+import { InterceptTriggerService } from './services/intercept_trigger';
+import { InterceptUserInteractionService } from './services/intercept_user_interaction';
+import { TRIGGER_INFO_API_ROUTE } from '../common/constants';
+import type { TriggerInfo } from '../common/types';
+
+interface InterceptTriggerRouteContext extends RequestHandlerContext {
+  triggerInfo: Promise<TriggerInfo>;
+}
+
+interface InterceptTriggerCoreSetup {
+  kibanaVersion: string;
+}
+
+export class InterceptsTriggerCore {
```

Comment:
Naming is hard but we've found using the term 'core' when referring to anything other than Core in `src/core` can be confusing. Take Reporting for example: https://github.com/elastic/kibana/blob/12b7429afe0fb470fa6c85168c9040a5957bbe8b/x-pack/platform/plugins/private/reporting/server/index.ts#L25